deepcell/applications/application_test.py

```diff
@@ -243,13 +243,27 @@ def _format_model_output(Lx):
         y = app._format_model_output(x)
         self.assertAllEqual(x, y['inner-distance'])
 
+    def test_batch_predict(self):
+        for num_images in [4, 8, 10]:
+            for num_pred_heads in [1, 2]:
+                model = DummyModel(n_out=num_pred_heads)
+                app = Application(model)
+
+                x = np.random.rand(num_images, 128, 128, 1)
+                y = app._batch_predict(x, batch_size=4)
+                self.assertEqual(x.shape, y[0].shape)
```

Comment:
I think a good test here would be to use [`pytest-mock` to "spy"](https://github.com/pytest-dev/pytest-mock/#spy) on `model.predict` (or another function) and make sure it's called the correct number of times.

Reply:
I'm running into issues getting mocker declared in the test function, since it's a method of a test class. 
When I do 
```
    def test_batch_predict(self, mocker):
        for num_images in [4, 8, 10]: 
```

I get the following error message about a missing argument in test_batch_predict when pytest is trying to call the method. I'm guessing I need to use a decorator or something instead?

![image](https://user-images.githubusercontent.com/13770365/128735875-6e7491fd-fc43-4ec9-91c7-3b048106090a.png)

Reply:
Oh, I forgot that TensorFlow uses `unittest` instead. Try using [the `mock` library](https://docs.python.org/3/library/unittest.mock.html#the-mock-class) instead